src/types/resolveDescriptors.ts

```diff
@@ -862,6 +862,13 @@ export function resolveDescriptors(ctx: CompilerContext, Ast: FactoryAst) {
             }
         }
 
+        if (a.kind === "asm_function_def" && isGetter) {
+            throwCompilationError(
+                "Assembly functions cannot be getters",
```

Comment:
This is a confusing error message: any module-level function cannot be a getter, since it's not a contract function

Reply:
Then let's just reuse the `Getters must be defined within a contract` error message that we give for getters outside contracts

Reply:
And also, let's add a test with an `asm` getter inside a contract

Reply:
Added a test, it seems the current parser can't parse the asm getter correctly. AFAIK, we already have an issue about this

Reply:
Could you link that issue here, please?

Reply:
Maybe this one #1619?

Reply:
> Maybe this one #1619?

yep